src/test/UniswapV3SplitRouteCallee.t.sol

```diff
@@ -0,0 +1,359 @@
+            hevm.store(token, keccak256(abi.encode(uint256(i), address(this))), bytes32(amount));
+            if (GemAbstract(token).balanceOf(address(this)) == amount) {
+                // Found it
+                return;
+            } else {
+                // Keep going after restoring the original value
+                hevm.store(token, keccak256(abi.encode(uint256(i), address(this))), prevValue);
+            }
+        }
+    }
+
+    address linkAddr;
+    address daiAddr;
+    address vatAddr;
+    address linkJoinAddr;
+    address daiJoinAddr;
+    address dogAddr;
+    address jugAddr;
+    address linkClipAddr;
+    address linkPipAddr;
+
+    Hevm hevm;
+    GemAbstract     link;
+    VatAbstract     vat;
+    DaiAbstract     dai;
+    GemJoinAbstract linkJoin;
+    DogAbstract     dog;
+    JugAbstract     jug;
+    ClipAbstract    linkClip;
+
+    OsmAbstract linkPip;
+
+    function setAddresses() private {
+        ChainlogHelper helper = new ChainlogHelper();
+        ChainlogAbstract chainLog = helper.ABSTRACT();
+        linkAddr = chainLog.getAddress('LINK');
+        vatAddr = chainLog.getAddress('MCD_VAT');
+        daiAddr = chainLog.getAddress('MCD_DAI');
+        linkJoinAddr = chainLog.getAddress('MCD_JOIN_LINK_A');
+        daiJoinAddr = chainLog.getAddress('MCD_JOIN_DAI');
+        dogAddr = chainLog.getAddress('MCD_DOG');
+        jugAddr = chainLog.getAddress('MCD_JUG');
+        linkClipAddr = chainLog.getAddress('MCD_CLIP_LINK_A');
+        linkPipAddr = chainLog.getAddress('PIP_LINK');
+    }
+
+    function setInterfaces() private {
+        hevm = Hevm(hevmAddr);
+        link = GemAbstract(linkAddr);
+        vat = VatAbstract(vatAddr);
+        dai = DaiAbstract(daiAddr);
+        linkJoin = GemJoinAbstract(linkJoinAddr);
+        dog = DogAbstract(dogAddr);
+        jug = JugAbstract(jugAddr);
+        linkClip = ClipAbstract(linkClipAddr);
+        linkPip = OsmAbstract(linkPipAddr);
+    }
+
+    VaultHolder          ali;
+    UniswapV3SplitCallee dan;
+
+    address aliAddr;
+    address danAddr;
+    address UniswapV3Router2;
+    bytes   uniswapTxDataProfit;
+    bytes   uniswapTxDataNoProfit;
+
+    function getPermissions() private {
+        hevm.store(dogAddr, keccak256(abi.encode(address(this), uint256(0))), bytes32(uint256(1)));
+        hevm.store(vatAddr, keccak256(abi.encode(address(this), uint256(0))), bytes32(uint256(1)));
+        hevm.store(linkPipAddr, keccak256(abi.encode(address(this), uint256(0))), bytes32(uint256(1)));
+        linkPip.kiss(address(this));
+    }
+
+    function setEnvVars() private {
+        // Execute this test with fresh data via
+        // `(cd scripts/uniswap-split-route-callee && npm ci && node index.js)`
+        // check the script for more details on how to use universal router
+        UniswapV3Router2 = hevm.envOr('UniswapV3Router2', 0x68b3465833fb72A70ecDF485E0e4C7bD8665Fc45);
+        uniswapTxDataNoProfit = hevm.envOr(
```

Comment:
How was the no profit vs profit amount generated?
Couldn't get it from looking at the script as well.

I guess I also don't really get why the multicall data should be different and separated to cases, as there is only one best uniswap price at a given time (and the min profit amount is enforced in the callee).
Maybe I'm missing something.

Reply:
> How was the no profit vs profit amount generated?

via using the different `linkAmountToSwap` values.

- if we derive this amount from 1.5*dust (in link equivalent): the this will be "no profit" for the coded test scenario
- if we derive this from 2*dust: this is the case where it has profit.

i.e. script function `getLinkAmountToSwap(2, 1)` yields us the case with sufficient profit, `getLinkAmountToSwap(3, 2)` - with no profit.

for "why this happens this way? why can't we use same link amounts?" see the bottom-most section 

> as there is only one best uniswap price at a given time

that is true, but for different amounts of input currency the return of the `AlphaRouter` can potentially be different (e.g. in distribution if currency among routes, routes themselves) as of [the description](https://uniswap.org/blog/auto-router)

___

Why different link amounts?

during development of the no-profit test the exchange with same calldata always resulted into profitable case. So this required to fine-tune the swapped amounts.
With same calldata (same link amount) as in profit case taking the auction as soon as it stops being non-profitable still results into profit. Making the link amount smaller resolved this problem.

Reply:
I think if it makes your life much easier you can remove the no profit test.

That's maybe actually cleaner than purposely swapping only part of the taken auctioned collateral so there will be not much dai profit. That part is really confusing as it's different than all the other tests in this repo.

Reply:
removed the no-profit test